docs: Add frontend engineering section and table styling

tech-andgar opened the pull request:
## Summary by Sourcery

Add a new “Frontend Engineering” section to the documentation with two in-depth lifecycle guides and a Spanish translation, and enhance markdown table styling with improved fonts, borders, sticky headers/columns, and mobile responsiveness.

New Features:
- Introduce “Frontend Engineering” sidebar entry and directory for new docs
- Add detailed component lifecycle examples MDX covering React, Angular, Vue, Flutter, Swift, Kotlin, HTMX, Astro, and Svelte
- Add cross-stack developer guide MDX on component lifecycle philosophies and hooks

Enhancements:
- Enhance markdown table styling in custom CSS with font adjustments, borders, sticky headers/columns, and responsive media queries

Documentation:
- Include Spanish translation of the Flutter StatefulWidget lifecycle and performance tutorial

Chores:
- Fix minor trailing comma formatting in sidebar config

tech-andgar committed:
docs(frontend-engineering): add component lifecycle guide across frameworks

Add comprehensive documentation comparing component lifecycle patterns in React, Angular, Vue, Flutter, Swift, Kotlin, HTMX, Astro, and Svelte. The guide explains core lifecycle phases and their implementation across different frameworks to help developers understand fundamental UI development concepts.

tech-andgar committed:
docs(frontend-engineering): add practical Component Lifecycle Examples - React, Angular, Vue, Flutter & More

tech-andgar committed:
docs(Flutter): add Spanish tutorial on StatefulWidget lifecycle and performance

Add comprehensive documentation explaining StatefulWidget lifecycle, state management, and performance considerations in Flutter. The tutorial uses actor/role analogy to simplify complex concepts and includes practical code examples demonstrating state preservation with GlobalKey.

tech-andgar committed:
feat(sidebar): add frontend engineering section to sidebar

Add new sidebar item for frontend engineering content to improve navigation and content organization

tech-andgar committed:
style(css): improve table readability and responsive layout

- Add responsive table styles with horizontal scroll for better mobile experience
- Adjust column widths and padding for improved readability
- Make first column sticky for better table navigation
- Optimize styles for different screen sizes

tech-andgar commented:
@sourcery-ai title

tech-andgar committed:
docs(frontend): add component lifecycle guide across modern frameworks

Add comprehensive developer guide explaining component lifecycle concepts
across React, Angular, Vue, Flutter, Swift, Kotlin, HTMX, Astro and Svelte.
The guide covers core lifecycle phases, framework-specific implementations,
and patterns for stateless components.

miladvafaeifard commented:
<img width="1037" alt="image" src="https://github.com/user-attachments/assets/f8a087e8-ba08-4d84-b29c-d15455fd8488" />

<img width="1343" alt="image" src="https://github.com/user-attachments/assets/a44af74d-ccd1-4882-b978-b766b3866298" />

Responsive seems not working well

tech-andgar committed:
chore: update cSpell dictionary with new tech-related terms

Add new terms related to Android, Kotlin, iOS, and web development to the cSpell dictionary to prevent false positives in spell checking. Includes terms like 'androidx', 'Jetpack', 'kotlinx', 'lateinit', 'htmx', and 'Subview'.

tech-andgar committed:
style(css): adjust sidebar and container widths for better responsiveness

Update left position of right sidebar from 78vw to 80vw and change max-width of container from 68rem to 60vw to improve layout responsiveness across different screen sizes

tech-andgar committed:
style(css): improve responsive layout and table text wrapping

- Adjust main pane width to be responsive across different screen sizes
- Replace word-wrap with text-wrap: balance for better table text rendering

tech-andgar committed:
style(css): adjust layout widths and padding for better responsiveness

- Increase main pane width from 55vw to 55.5vw for 1024px breakpoint
- Add new 1600px breakpoint with 68vw main pane width
- Add consistent padding to content panel
- Widen tables to 98% width for better content display

tech-andgar committed:
docs(frontend-engineering): reorder framework columns in lifecycle comparison table

Reorganize the framework columns to group web frameworks (React, Angular, Vue) first, followed by HTMX, Astro, Svelte, then mobile/native frameworks (Flutter, Swift, Kotlin) for better readability and logical grouping
```

The commit:
1. Uses `docs` type since it's documentation update
2. Specifies scope as `frontend-engineering`
3. Briefly describes the column reordering change
4. Explains the reasoning in the body (grouping related frameworks together) without repeating the subject line

tech-andgar committed:
style(css): improve table styling and responsiveness

- Simplify table layout by removing redundant width properties
- Enhance sticky header and first column behavior with better z-index values
- Adjust border styling and spacing for better visual consistency

tech-andgar commented:
@sourcery-ai dismiss

tech-andgar committed:
style(css): adjust table styles for better readability

- Change table height to max-height for better content handling
- Remove redundant sticky position selector
- Increase min-width of non-first columns for better content display

tech-andgar committed:
docs(frontend-engineering): update React lifecycle section with class components

Add detailed explanation of React class component lifecycle methods alongside hooks
to provide historical context and complete picture of React's evolution
